third_party/accessibility/base/string_utils.cc

```diff
@@ -18,6 +20,48 @@
 
 namespace base {
 
+using double_conversion::DoubleToStringConverter;
+using double_conversion::StringBuilder;
+
+namespace {
+static char const kExponentChar = 'e';
+static const char* const kInfinitySymbol = "Infinity";
+static const char* const kNaNSymbol = "NaN";
+
+// The number of digits after the decimal we allow before switching to
+// exponential representation.
+constexpr int kDecimalInShortestLow = -6;
+// The number of digits before the decimal we allow before switching to
+// exponential representation.
+constexpr int kDecimalInShortestHigh = 12;
+constexpr int kConversionFlags =
+    DoubleToStringConverter::EMIT_POSITIVE_EXPONENT_SIGN;
+
+static const double_conversion::DoubleToStringConverter*
+GetDoubleToStringConverter() {
+  static DoubleToStringConverter converter(
+      kConversionFlags, kInfinitySymbol, kNaNSymbol, kExponentChar,
+      kDecimalInShortestLow, kDecimalInShortestHigh, 0, 0);
+  return &converter;
+}
+
+std::string NumberToStringImpl(double number, bool is_single_precision) {
+  if (number == 0.0) {
+    return "0";
+  }
+  constexpr int kBufferSize = 128;
+  char char_buffer[kBufferSize];
```

Comment:
Consider `std::array<char>` since the size is a constexpr.

Reply:
LOL, I had it that way, and converted to this because it seemed less complex (and no benefit to using the std::array).